<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>singlecell-clustering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singlecell-clustering</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">singlecell-clustering</h1>

</div>


<p>We explore the clustering results using CountClust on the single cell RNA seq data to identify mechanism of Merkel Cell Carcinoma response and escape from T Cell Immunotherapy [PI: Aude Chapuis].</p>
<p>The data is obtained from a single MCPyV+-MCC patient with widely metastatic MCC, who was treated, and the treatment worked in the initial phase of one year, but then it relapsed. The single cell data was obtained using Drop-seq method from 10X Genomics. 10X-Genomics technology is uniquely poised to provide a detailed multidimensional view of the tumor microenvironment and cellular response. Key material from time points of response and progression including viable frozen single cell tumor digests (contain tumor &amp; stroma &amp; infiltrating immune cells), TIL cultures and PBMCs.</p>
<p>Overall the data is collected over 4 time periods (August 2013, February 2015, February 2016 and October 2016). August 2013 is around the time of cancer detection, February 2015 is when the treatment is started, which is continued for one year, February 2016 is towards the end of the treatment period when the treatment is found to be working. Post this, the cancer relapses and October 2016 corresponds to a time period by which time the cancer has relapsed.</p>
<p>We consider the clustering of the single cells for each of the four time periods.</p>
<ul>
<li><p><a href="pbmc_MCC1050.html">CountClust + tSNE analysis of scRNA-seq data from Aug 2013 - MCC1050</a></p></li>
<li><p><a href="pbmc_GRN304.html">CountClust + tSNE analysis of scRNA-seq data from Feb 2015 - GRN0304</a></p></li>
<li><p><a href="pbmc_GRN535.html">CountClust + tSNE analysis of scRNA-seq data from Feb 2016 - GRN0535</a></p></li>
<li><p><a href="pbmc_GRN760.html">CountClust + tSNE analysis of scRNA-seq data from Aug 2013 - GRN0760</a></p></li>
</ul>
<p>We also pool the single cell data from the four time periods and perform the clustering.</p>
<ul>
<li><a href="aggregated_PBMC.html">CountClust + tSNE analysis of scRNA-seq data from all 4 time periods</a></li>
</ul>
<p>Based on the above studies, we note down the summary of the results obtained.</p>
<ul>
<li></li>
</ul>
<div id="summary-of-the-results-from-the-analysis" class="section level2">
<h2>Summary of the results from the analysis</h2>
<p>We check for the UMI patterns and dropout rates compared to the characteristics of the genes like gene length, GC content etc.</p>
</div>
<div id="exploration-of-scrna-seq-data---umi-dropouts-etc." class="section level2">
<h2>Exploration of scRNA-seq data - UMI, dropouts etc.</h2>
<p><a href="umi_study_TFH.html">UMI study for TFH146</a></p>
<p><a href="dropout_study.html">Dropout rates versus position of the genes along chromosome and pathway groups</a></p>
<p><a href="dropout_gc_study.html">Dropout rates versus GC content of the genes</a></p>
<p>We next perform an analysis of the dispersion of genes against the average expression using Seurat and do a pathway based enrichment analysis of the genes that seem to be outliers in this plot.</p>
<p><a href="cvplot_analysis.html">Dispersion versus Average expression analysis</a></p>
</div>
<div id="countclust-analysis-of-the-joint-ln-and-pbmc-cells-for-live-cells-and-tfh-sorted-cells" class="section level2">
<h2>CountClust analysis of the joint LN and PBMC cells for Live cells and TFH sorted cells</h2>
<p>We apply CountClust Grade of Membership (GoM) model for clustering the single cell data from lymph node (LN) and peripheral blood mononuclear cells (PBMC) of three HIV- and one HIV+ patients.</p>
<p>An example of countclust fit on single cell data from one patient (TFH146) on just the Live cells.</p>
<p><a href="ex_countclust.html">Example application of CountClust</a></p>
<p><a href="test_script.html">CountClust on TFH146 scRNA-seq data</a></p>
<p><a href="tfh041Explore.html">CountClust on TFH041 scRNA-seq data</a></p>
<p><a href="tfh095Explore.html">CountClust on TFH095 scRNA-seq data</a></p>
<p><a href="mp046Explore.html">CountClust on MP046 scRNA-seq data</a></p>
</div>
<div id="separate-countclust-analysis-of-the-ln-and-pbmc-cells" class="section level2">
<h2>Separate CountClust analysis of the LN and PBMC cells</h2>
<p>Separate Grade of Membership model fit and subsequent gene annotations and gene set enrichment analysis of the clusters, for the different patients</p>
<p>First, we perform this analysis for TFH146</p>
<p><a href="separate_countclust_LN_PBMC_Live_TFH146.html">CountClust application separately on LN and PBMC Live cells for TFH146</a></p>
<p><a href="separate_countclust_LN_PBMC_TFH_TFH146.html">CountClust application separately on LN and PBMC TFH cells for TFH146</a></p>
<p>Next we perform the analysis for TFH041</p>
<p><a href="separate_countclust_LN_PBMC_Live_TFH041.html">CountClust application separately on LN and PBMC Live cells for TFH041</a></p>
<p><a href="separate_countclust_LN_PBMC_TFH_TFH041.html">CountClust application separately on LN and PBMC TFH cells for TFH041</a></p>
<p>Finally the analysis on the Live cells for the TFH095 and MP046</p>
<p><a href="separate_countclust_LN_PBMC_Live_TFH095.html">CountClust application separately on LN and PBMC Live cells for TFH095</a></p>
<p><a href="separate_countclust_LN_PBMC_Live_MP046.html">CountClust application separately on LN and PBMC Live cells for MP046</a></p>
</div>
<div id="comparison-of-countclust-results-with-t-sne" class="section level2">
<h2>Comparison of CountClust results with t-SNE</h2>
<p>We projected the expression of top driving genes from fitting the CountClust model on the t-SNE space obtained using the Seurat package by Satijalab.</p>
<p>The projections for TFH146, TFH041 and TFH095 are</p>
<p><a href="tsne_countclust_TFH146.html">t-SNE projection of CountClust top cluster driving genes - TFH 146</a></p>
<p><a href="tsne_countclust_TFH095.html">t-SNE projection of CountClust top cluster driving genes - TFH 095</a></p>
<p><a href="tsne_countclust_TFH041.html">t-SNE projection of CountClust top cluster driving genes - TFH 041</a></p>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
