<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>singlecell-clustering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singlecell-clustering</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">singlecell-clustering</h1>

</div>


<div id="merkel-cell-carcinoma-data-study" class="section level2">
<h2>Merkel Cell Carcinoma Data Study</h2>
<p>We explore the clustering results using CountClust on the single cell RNA seq data to identify mechanism of Merkel Cell Carcinoma response and escape from T Cell Immunotherapy [PI: Aude Chapuis].</p>
<p>The data is obtained from a single MCPyV+-MCC patient with widely metastatic MCC, who was treated, and the treatment worked in the initial phase of one year, but then it relapsed. The single cell data was obtained using Drop-seq method from 10X Genomics. 10X-Genomics technology is uniquely poised to provide a detailed multidimensional view of the tumor microenvironment and cellular response. Key material from time points of response and progression including viable frozen single cell tumor digests (contain tumor &amp; stroma &amp; infiltrating immune cells), TIL cultures and PBMCs.</p>
<p>Overall the data is collected over 4 time periods (August 2013, February 2015, February 2016 and October 2016). August 2013 is around the time of cancer detection, February 2015 is when the treatment is started, which is continued for one year, February 2016 is towards the end of the treatment period when the treatment is found to be working. Post this, the cancer relapses and October 2016 corresponds to a time period by which time the cancer has relapsed.</p>
<p>We consider the clustering of the single cells for each of the four time periods.</p>
<ul>
<li><p><a href="pbmc_MCC1050.html">CountClust + tSNE analysis of scRNA-seq data from Aug 2013 - MCC1050</a></p></li>
<li><p><a href="pbmc_GRN304.html">CountClust + tSNE analysis of scRNA-seq data from Feb 2015 - GRN0304</a></p></li>
<li><p><a href="pbmc_GRN535.html">CountClust + tSNE analysis of scRNA-seq data from Feb 2016 - GRN0535</a></p></li>
<li><p><a href="pbmc_GRN760.html">CountClust + tSNE analysis of scRNA-seq data from Oct 2016 - GRN0760</a></p></li>
</ul>
<p>We also pool the single cell data from the four time periods and perform the clustering.</p>
<ul>
<li><a href="aggregated_PBMC.html">CountClust + tSNE analysis of scRNA-seq data from all 4 time periods</a></li>
</ul>
<p>We perform Gene Set Enrichment Analysis (GSEA) of the cluster driving gene sets obtained from the cluster model fits from each time period and also for aggregated data from 4 periods.</p>
<ul>
<li><p><a href="gsea_MCC1050.html">GSEA of CountClust clusters from Aug 2013 - MCC1050</a></p></li>
<li><p><a href="gsea_GRN304.html">GSEA of CountClust clusters from Feb 2015 - GRN0304</a></p></li>
<li><p><a href="gsea_GRN535.html">GSEA of CountClust clusters from Feb 2016 - GRN0535</a></p></li>
<li><p><a href="gsea_GRN760.html">GSEA of CountClust clusters from Oct 2016 - GRN0760</a></p></li>
<li><p><a href="gsea_aggregated.html">GSEA of CountClust clusters from all 4 time periods</a></p></li>
</ul>
<p>Based on the above studies, we note down the summary of the results obtained.</p>
<div id="summary-of-the-results-from-the-analysis" class="section level3">
<h3>Summary of the results from the analysis</h3>
<ul>
<li><p>A common pattern observed for each of the individual CountClust analysis done at each of the 4 time periods is that for <span class="math inline">\(K=3\)</span>, one cluster seems to be driven by the myeloid markers like the S100* group, another by the hemoglobin expressed genes and finally a cluster that is driven by the cancer related genes. We however see the proportional representation of these clusters to vary from one time period to the other.</p></li>
<li><p>For higher order clusters, we sometimes get additional clusters to be driven by the cytotoxic T cell related genes (GZMA, GZMH) and NK cell related genes (NKG7). Also for further higher order clusters, PF4 or megakaryocyte gene related clusters are also observed. Plus, higher clusters also include clusters driven by CD74, CD79 and CD79A, which are markers for B cells. We also find a close connection of known markers for different cell types with the different higher order clusters we are observing (PF4, CD79A, CD74, S100A9, S100A8 etc).</p></li>
<li><p>For MCC1050, we find very low proportional representation of the hemoglobin related genes in the cells. This is at early stages of tumor progression before the T cells have been infused. Note that the samples at this stage are a combination of PBMCs, TILs and sc tumor digest.</p></li>
<li><p>For GRN304, we see a clearer separation of the clusters, and the hemoglobin related cluster seems to now occur in a handful of cells (much higher proportion compared to the first time). The samples at this stage are PBMCs.</p></li>
<li><p>For GRN535, which is taken towards the end of the treatment period, we see a great reduction in the representation of the hemoglobin gene driven cluster. The myeloid cells cluster and the one driven by MALAT1, B2M dominate most tof the cells at this stage. It is known that radiation therapy can cause low levels of RBCs and WBCs resulting in low blood cell counts. That could be a possible reason why the hemoglobin genes are expressed in a small proportion of cells at this stage. The samples at this stage are PBMCs.</p></li>
<li><p>During the final stage when the cancer has relapsed, we see the proportion of hemoglobin expressed genes has increased again. The samples at this stage are a combination of PBMCss, TILs and sc tumor digests.</p></li>
<li><p>When we aggregate the data across the 4 time points, we fortunately do not see much bias due to batch effects resulting from cell collection and sequencing at these different time points. Because the data is clean, a lot of the features we see in the clusters in each of these time periods, is also visible for the pooled data across the 4 time periods.</p></li>
<li><p>Even for the combined data, for <span class="math inline">\(K=3\)</span>, the groups are similar to the individual time period data. For <span class="math inline">\(K=6\)</span>, we get a cluster determined by CD79A, CD79B and CD74. This could be the B cell receptor cells. Also the 1st cluster seems to be driven by cancer related genes and markers for NK cells.Also, other clusters include myeloid lineage cells (S100*) and the hemoglobin related cells (likely RBCs).</p></li>
<li><p>The GSEA analysis (done using <a href="http://cpdb.molgen.mpg.de/">Consensus pathDB database</a>) seems to identify heme related pathways for the hemoglobin driving genes cluster and an immunity genes driven cluster for smaller clusters. But for larger clusters <span class="math inline">\(K=6\)</span>, we do get clusters that are driven by T cell pathways, NK cell related pathways, toll like receptors and myeloid cell pathways and B cell pathways.</p></li>
</ul>
</div>
</div>
<div id="hiv-data-study" class="section level2">
<h2>HIV Data study</h2>
<p>We next apply countClust and project the results on tSNE plot for three HIV+ and one HIV- patients. The single cell data were collected from Live cells and TF sorted cells. Also, for each case, the cells were obtained either from lymph node (LN) or from PBMCs.</p>
<p>We first look at results from application of CountClust separately on the LN and PBMC data from Live cells and TF sorted cells from each patient.</p>
<ul>
<li><p><a href="tsne_countclust_TFH146_Live.html">CountClust + tSNE analysis of scRNA-seq data from LN and PBMC data in Live cells in TFH146</a></p></li>
<li><p><a href="tsne_countclust_TFH041_Live.html">CountClust + tSNE analysis of scRNA-seq data from LN and PBMC data in Live cells in TFH041</a></p></li>
<li><p><a href="tsne_countclust_TFH095_Live.html">CountClust + tSNE analysis of scRNA-seq data from LN and PBMC data in Live cells in TFH095</a></p></li>
<li><p><a href="tsne_countclust_MP046_Live.html">CountClust + tSNE analysis of scRNA-seq data from LN and PBMC data in Live cells in MP046</a></p></li>
<li><p><a href="tsne_countclust_TFH146_TF.html">CountClust + tSNE analysis of scRNA-seq data from LN and PBMC data in TF sorted cells in TFH146</a></p></li>
<li><p><a href="tsne_countclust_TFH041_TF.html">CountClust + tSNE analysis of scRNA-seq data from LN and PBMC data in TF sorted cells in TFH041</a></p></li>
</ul>
<p>We next combine the LN and PBMC data and apply CountClust and visualize by tSNE the results.</p>
<p>First we do it for the Live cells</p>
<ul>
<li><p><a href="tsne_countclust_TFH095_Live_pooled.html">CountClust + tSNE analysis of scRNA-seq data from LN + PBMC data in Live cells in TFH095</a></p></li>
<li><p><a href="tsne_countclust_TFH146_Live_pooled.html">CountClust + tSNE analysis of scRNA-seq data from LN + PBMC data in Live cells in TFH146</a></p></li>
<li><p><a href="tsne_countclust_TFH041_Live_pooled.html">CountClust + tSNE analysis of scRNA-seq data from LN + PBMC data in Live cells in TFH041</a></p></li>
<li><p><a href="tsne_countclust_MP046_Live_pooled.html">CountClust + tSNE analysis of scRNA-seq data from LN + PBMC data in Live cells in MP046</a></p></li>
</ul>
<p>Next we do the same for the TF sorted cells</p>
<ul>
<li><p><a href="tsne_countclust_TFH146_TF_pooled.html">CountClust + tSNE analysis of scRNA-seq data from LN + PBMC data in TF sorted cells in TFH146</a></p></li>
<li><p><a href="tsne_countclust_TFH041_TF_pooled.html">CountClust + tSNE analysis of scRNA-seq data from LN + PBMC data in TF sorted cells in TFH041</a></p></li>
<li><p><a href="tsne_countclust_MP046_TF_pooled.html">CountClust + tSNE analysis of scRNA-seq data from LN + PBMC data in TF sorted cells in MP046</a></p></li>
</ul>
<p>We also check for the individual specific effects in driving the clusters by merging the TF sorted cells from two individuals together and then running CountClust on the merged data.</p>
<ul>
<li><a href="tsne_countclust_TF_TFH041_TFH146.html">CountClust + tSNE analysis of scRNA-seq data from LN + PBMC data in TF sorted cells in TFH041+TFH146</a></li>
</ul>
<p>We perform Gene Set Enrichment Analysis (GSEA) of the cluster driving gene sets obtained from the cluster model fits in some of the studies above.</p>
<ul>
<li><a href="gsea_HIV.html">GSEA of the cluster driving gene sets in HIV studies</a></li>
</ul>
<div id="summary-of-the-results-from-the-analysis-1" class="section level3">
<h3>Summary of the results from the analysis</h3>
<ul>
<li><p>For TFH146, we find that in the LN cells, there is a cluster pattern driven by CD74 and other HLA related genes. The same is not observed for the PBMC cells, but the latter shows some clustering due to myeloid markers like S100*.</p></li>
<li><p>For TFH095, in LN cells, we see the HLA driven cluster, as in TFH146. On the other hand for PBMCs, for <span class="math inline">\(K=3\)</span>, we see a cluster driven by cytotoxic cells and NK cells and then a cluster that is driven by a mix of HLA and S100* genes. My feeling is that we will get a further split for higher <span class="math inline">\(K\)</span>.</p></li>
<li><p>So many HLA genes driven cluster was not observed for the cancer study as is in this case.</p></li>
<li><p>For TF sorted cells in TFH146, no clear clustering pattern is observed from the t-sne plot in case of the LN cells and PBMC cells.</p></li>
<li><p>For TF sorted cells in LN in TFH041, one cluster is dominated by HLA genes, anotherby CD3D, CCL5, CD48 etc. These are the two clear clusters observed in this data.</p></li>
<li><p>When we combine the LN and PBMC cells, the first distinct grouping is between the LN cells and the PBMC cells.</p></li>
<li><p>In TFH095, for <span class="math inline">\(K=3\)</span>, the three clusters are formed as follows - one by HLA genes, one by NK related genes, GZMA, GZMH, GNLY etc, and the third by markers like CD3D,CD2, CM1 etc</p></li>
<li><p>Also when cells from two individuals are mixed (TF sorted cells in this case from TFH041 and TFH146), then the first clustering is at the level of the individual as checked by the last script.</p></li>
<li><p>The GSEA analysis (done using <a href="http://cpdb.molgen.mpg.de/">Consensus pathDB database</a>) of the clustering analysis of seprate CountClust application on LN and PBMC cells for Live cells or PBMC cells, seems to segregate for K=2 two clusters driven by APCs (antigen presenting cells) and T cells driven cluster, while ofr K=3, it seems to pick up NK cell related clusters in most cases.</p></li>
</ul>
</div>
</div>
<div id="exploration-of-the-umis" class="section level2">
<h2>Exploration of the UMIs</h2>
<p>The data in both the MCC and the HIV studies are UMI data. We checked for possible biases in the UMI assignment which would affect the counts of the UMIs and may require further normalization. We look first at how the number of UMIs assigned to a gene changes based on the properties of the gene - namely, the number of TSS, the GC content, the length of the gene etc.</p>
<p>Previous studies tend to indicate that unlike the read counts, the UMI counts are stable with respect to the gene length. This is good for CountClust, as the latter does not notmalize the data by gene lengths and only performs the library size normalization. This would suggest that CountClust should adapt better in handling UMI data compared to non UMI data for single cell studies.</p>
<ul>
<li><p><a href="umi_study_TFH.html">UMI study for TFH146</a></p></li>
<li><p><a href="umi_study_2.html">UMI versus Transcription Start Sites of the genes</a></p></li>
<li><p><a href="umi_study_3.html">UMI versus Transcript count and Transcript length of the genes</a></p></li>
<li><p><a href="dropout_study.html">Dropout rates versus position of the genes along chromosome and pathway groups</a></p></li>
<li><p><a href="dropout_gc_study.html">Dropout rates versus GC content of the genes</a></p></li>
</ul>
<p>We next perform an analysis of the dispersion of genes against the average expression using Seurat and do a pathway based enrichment analysis of the genes that seem to be outliers in this plot.</p>
<ul>
<li><a href="cvplot_analysis.html">Dispersion versus Average expression analysis</a></li>
</ul>
<p>Finally, we have seen that usually there is a bias in the base composition of the UMIs that get added to the reads which are finally mapped. This seemed to be a big issue in the Fluidigm data we looked at before, but we wanted to check if they indeed play a role even for this data. The results summarized below suggest that the UMI bias is not appreciable for this data.</p>
<ul>
<li><a href="umi_composition.html">UMI base composition study : 10X Genomics vs Fluidigm</a></li>
</ul>
<p>Overall we did not see a relationship between the UMI patterns and the gene structure, we found very little compositional bias specially at the first few sites.</p>
</div>
<div id="x-genomics-pbmcsorted-unsorted-data" class="section level2">
<h2>10X genomics PBMC(sorted + unsorted data)</h2>
<p>We applied CountClust on the sorted immune cells data and unsorted PBMC data from 10X genomics. The results for the sorted cells alone study is the following</p>
<ul>
<li><a href="sorted_cells_10X_analysis.html">sorted immune cells CountClust study</a></li>
</ul>
<p>The results from applying countClust on just the unsorted PBMC data are</p>
<ul>
<li><a href="pbmc_68k.html">unsorted PBMC cells CountClust study</a></li>
</ul>
<p>The results from applying CountClust on the sorted cells + unsorted PBMC cells is the following</p>
<ul>
<li><a href="pooled_sorted_unsorted_pbmc_10X_analysis.html">CountClust study on sorted immune cells + unsorted PBMC cells</a></li>
</ul>
<p>Since we have data available on sorted cells, and from the above analysis, merging the unsorted cells with the sorted ones seemed to preserve the biological structure, we decided to apply a supervised version of the Countclust method or grade of membership method and also pit it against a more standard classification algorithm like Support Vector Machines (SVM).</p>
<p>The results from a supervised Countclust model fit and a partially supervised model fits are presented below.</p>
<ul>
<li><p><a href="classification_gom.html">supervised Countclust study on sorted + unsorted PBMC cells</a></p></li>
<li><p><a href="classification_gom_2.html">partially supervised Countclust study on sorted + unsorted PBMC cells, allowing for 2 more clusters</a></p></li>
<li><p><a href="svm_classtpx_comparison.html">Comparison of SVM and Countclust results on sorted + unsorted PBMC data</a></p></li>
</ul>
<p>We show below that the training accuracy of the hard classification from classtpx is better than that of Support Vector Machines (SVM).</p>
<ul>
<li><a href="train_accuracy_svm_classtpx.html">Training accuracy of classtpx and SVM on the sorted + unsorted PBMC data</a></li>
</ul>
<p>We next threshold the genes based on the divergence measure of the sorted cell gene expression to pick out the strongest driving genes for classification. As of now, since the divrgence measure tends to subside to 0 inside of the first 100 genes, we pick 100 strongest genes per class and so, for the 9 sorted cell types, we have a total of 900 genes, on which the thresholded analysis is done. The results are presented below</p>
<ul>
<li><p><a href="thresholded_classtpx.html">Thresholded classtpx performance on sorted + unsorted PBMC data</a></p></li>
<li><p><a href="train_accuracy_svm_classtpx_2.html">Training accuracy comparison of the original classtpx and the thresholded classtpx</a></p></li>
</ul>
<p>We also trained the model on the sorted immune cells from 10X Genomics and then observed the grades of memberships of the pooled MCC cells from the data on these classes. The results are presented below.</p>
<ul>
<li><a href="classification_gom_MCC.html">supervised CountClust study on the sorted immune cells + MCC cells pooled across 4 time points</a></li>
</ul>
<p>We also used Multivariate adaptive shrinkage (MASH) technique to detect the driving genes for each sorted cell type. We then project the expression patterns of those genes on the tSNE plot.</p>
<ul>
<li><a href="mash_input_prepare.html">MASH analysis - driving genes for the sorted immune cells</a></li>
</ul>
<p>We also observed if the sorted immune cells can be used to explain the grades of memberships in the bulk RNA-seq data - for example - the blood and blood tissues data in the GTEx analysis.</p>
<ul>
<li><a href="gtex_blood_sorted_immune.html">supervised Countclust study GTEx blood + sorted immune cells</a></li>
</ul>
<p>We look at more robust version of classtpx that uses trimmed mean at different degrees of trimming instead of mean of the expression patterns of cells for a particular class or cell type. We tried with median, but median did not seem to give good performance.</p>
<ul>
<li><p><a href="classification_gom_robust.html">robust supervised Countclust study on sorted + unsorted PBMC cells</a></p></li>
<li><p><a href="train_accuracy_svm_robust_classtpx.html">Training accuracy of robust classtpx at different degrees of trimming and SVM on the sorted + unsorted PBMC data</a></p></li>
<li><p><a href="svm_classtpx_comparison_robust.html">Comparison of SVM and robust supervised Countclust results on sorted + unsorted PBMC data</a></p></li>
</ul>
</div>
<div id="extras" class="section level2">
<h2>Extras</h2>
<p>The model for applying CountClust with covariates information is below</p>
<ul>
<li><a href="gom-cov.pdf">CountClust with covariate information : EM algorithm</a></li>
</ul>
<p>The results of plotting the divergence measure <span class="math inline">\(D\)</span> for the <em>ExtractTopFeatures()</em> function in <strong>CountClust</strong> is the following</p>
<ul>
<li><p><a href="score_analysis_countclust.html">Divergence measure steepness study in CountClust</a></p></li>
<li><p><a href="symmetry_check_qqplot.html">Testing whether QQplot can be used to select features in ExtractTopFeatures</a></p></li>
<li><p><a href="test_distrom.html">Testing for the speed of the distributed Multinomial regression - distrom package</a></p></li>
<li><p><a href="distrom_birds_data.html">Testing the performance of distrom and comparing with the linear model for birds ecological abundance data</a></p></li>
</ul>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
