---
title: 'GC content vs dropouts : TH146'
author: "Kushal K Dey"
date: "7/29/2017"
output: html_document
---

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE}
library(Seurat)
library(plyr)
library(dplyr)
library(Matrix)
library(cowplot)
library(stringr)
library(MAST)
library(doMC)
library(pheatmap)
library(grid)
library(stringr)
library(maptpx)
library(CountClust)
library(ggplot2)
library(GGally)
library(GSEABase)
library(limma)
library(reshape2)
library(data.table)
library(knitr)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(stringr)
library(NMF)
library(rsvd)
library(RColorBrewer)
library(MAST)
library(biomaRt)
```

## Processing the Data

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE}
###--- Live Cells
##- LN - no reads information
LiveCells_LN_UMI <- read.table('../data/TFH146/Live Cells/LN/reads.TFH_146_LN_S1_R1_001.fastq_bq10_star_corrected.umi.dge.txt', header = TRUE, row.names = 1)
LiveCells_LN_UMI_summary <- read.table('../data/TFH146/Live Cells/LN/reads.TFH_146_LN_S1_R1_001.fastq_bq10_star_corrected.umi.dge.summary.txt', header = TRUE, row.names = 1)
#- First filtering - remove cells with less than 10 genes
LiveCells_LN_UMI <- LiveCells_LN_UMI[, rownames(LiveCells_LN_UMI_summary)[which(LiveCells_LN_UMI_summary$NUM_GENES > 10)]]
LiveCells_LN_UMI_summary <- LiveCells_LN_UMI_summary[colnames(LiveCells_LN_UMI), ]

##- PBMC - no reads information
LiveCells_PBMC_UMI <- read.table('../data/TFH146/Live Cells/PBMC/reads.TFH_146_Blood_S2_R1_001.fastq_bq10_star_corrected.umi.dge.txt', header = TRUE, row.names = 1)
LiveCells_PBMC_UMI_summary <- read.table('../data/TFH146/Live Cells/PBMC/reads.TFH_146_Blood_S2_R1_001.fastq_bq10_star_corrected.umi.dge.summary.txt', header = TRUE, row.names = 1)
#- First filtering - remove cells with less than 10 genes
LiveCells_PBMC_UMI <- LiveCells_PBMC_UMI[, rownames(LiveCells_PBMC_UMI_summary)[which(LiveCells_PBMC_UMI_summary$NUM_GENES > 10)]]
LiveCells_PBMC_UMI_summary <- LiveCells_PBMC_UMI_summary[colnames(LiveCells_PBMC_UMI), ]



###--- TFH Cells
##- LN
TFH_LN_UMI <- read.table('../data/TFH146/TFH Sorted/LN/reads.TFH146_TFH_LN_S1_R1_001.fastq_bq10_star_corrected.umi.dge.txt.gz', header = TRUE, row.names = 1)
TFH_LN_reads <- read.table('../data/TFH146/TFH Sorted/LN/reads.TFH146_TFH_LN_S1_R1_001.fastq_bq10_star_corrected.reads.dge.txt.gz', header = TRUE, row.names = 1)
TFH_LN_UMI_summary <- read.table('../data/TFH146/TFH Sorted/LN/reads.TFH146_TFH_LN_S1_R1_001.fastq_bq10_star_corrected.umi.dge.summary.txt', header = TRUE, row.names = 1)
TFH_LN_reads_summary <- read.table('../data/TFH146/TFH Sorted/LN/reads.TFH146_TFH_LN_S1_R1_001.fastq_bq10_star_corrected.reads.dge.summary.txt', header = TRUE, row.names = 1)
#- First filtering - remove cells with less than 10 genes
TFH_LN_UMI <- TFH_LN_UMI[, rownames(TFH_LN_reads_summary)[which(TFH_LN_reads_summary$NUM_GENES > 10)]]
TFH_LN_reads <- TFH_LN_reads[, rownames(TFH_LN_reads_summary)[which(TFH_LN_reads_summary$NUM_GENES > 10)]]
TFH_LN_UMI_summary <- TFH_LN_UMI_summary[colnames(TFH_LN_UMI), ]
TFH_LN_reads_summary <- TFH_LN_reads_summary[colnames(TFH_LN_reads), ]

##- PBMC
TFH_PBMC_UMI <- read.table('../data/TFH146/TFH Sorted/PBMC/reads.TFH146_TFH_PBMC_S2_R1_001.fastq_bq10_star_corrected.umi.dge.txt.gz', header = TRUE, row.names = 1)
TFH_PBMC_reads <- read.table('../data/TFH146/TFH Sorted/PBMC/reads.TFH146_TFH_PBMC_S2_R1_001.fastq_bq10_star_corrected.reads.dge.txt.gz', header = TRUE, row.names = 1)
TFH_PBMC_UMI_summary <- read.table('../data/TFH146/TFH Sorted/PBMC/reads.TFH146_TFH_PBMC_S2_R1_001.fastq_bq10_star_corrected.umi.dge.summary.txt', header = TRUE, row.names = 1)
TFH_PBMC_reads_summary <- read.table('../data/TFH146/TFH Sorted/PBMC/reads.TFH146_TFH_PBMC_S2_R1_001.fastq_bq10_star_corrected.reads.dge.summary.txt', header = TRUE, row.names = 1)
#- First filtering - remove cells with less than 10 genes
TFH_PBMC_UMI <- TFH_PBMC_UMI[, rownames(TFH_PBMC_reads_summary)[which(TFH_PBMC_reads_summary$NUM_GENES > 10)]]
TFH_PBMC_reads <- TFH_PBMC_reads[, rownames(TFH_PBMC_reads_summary)[which(TFH_PBMC_reads_summary$NUM_GENES > 10)]]
TFH_PBMC_UMI_summary <- TFH_PBMC_UMI_summary[colnames(TFH_PBMC_UMI), ]
TFH_PBMC_reads_summary <- TFH_PBMC_reads_summary[colnames(TFH_PBMC_reads), ]
```

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE}
###--- Filtering datasets and merging
#- LN
identical(colnames(TFH_LN_UMI), rownames(TFH_LN_UMI_summary)) # TRUE
TFH_LN_UMI_filtered <- TFH_LN_UMI[, which(TFH_LN_UMI_summary$NUM_GENES >= 400 & TFH_LN_UMI_summary$NUM_TRANSCRIPTS >= 750)] # 344 cells
TFH_LN_UMI_filtered <- TFH_LN_UMI_filtered[which(apply(TFH_LN_UMI_filtered, 1, sum) != 0), ] # 14,856 genes and 344 cells
#- PBMC
identical(colnames(TFH_PBMC_UMI), rownames(TFH_PBMC_UMI_summary)) # TRUE
TFH_PBMC_UMI_filtered <- TFH_PBMC_UMI[, which(TFH_PBMC_UMI_summary$NUM_GENES >= 400 & TFH_PBMC_UMI_summary$NUM_TRANSCRIPTS >= 750)] # 361 cells
TFH_PBMC_UMI_filtered <- TFH_PBMC_UMI_filtered[which(apply(TFH_PBMC_UMI_filtered, 1, sum) != 0), ] # 14,657 genes and 361 cells
#- Merging
TFH_UMI <- rbind.fill(as.data.frame(t(TFH_LN_UMI_filtered)), as.data.frame(t(TFH_PBMC_UMI_filtered)))
rownames(TFH_UMI) <- c(paste(colnames(TFH_LN_UMI_filtered), 'LN', sep = '_'), paste(colnames(TFH_PBMC_UMI_filtered), 'PBMC', sep = '_'))
dim(TFH_UMI) # 705 cells and 17,772 genes
length(unique(c(rownames(TFH_LN_UMI_filtered), rownames(TFH_PBMC_UMI_filtered)))) # 19,384 genes
TFH_UMI[is.na(TFH_UMI)] <- 0 # NAs - replace NA by 0
TFH_UMI <- as.data.frame(t(TFH_UMI))
```

```{r echo=TRUE, eval=TRUE, message=FALSE, warning=FALSE}
###--- Filtering datasets and merging
#- LN
identical(colnames(LiveCells_LN_UMI), rownames(LiveCells_LN_UMI_summary)) # TRUE
LiveCells_LN_UMI_filtered <- LiveCells_LN_UMI[, which(LiveCells_LN_UMI_summary$NUM_GENES >= 400 & LiveCells_LN_UMI_summary$NUM_TRANSCRIPTS >= 750)] 
LiveCells_LN_UMI_filtered <- LiveCells_LN_UMI_filtered[which(apply(LiveCells_LN_UMI_filtered, 1, sum) != 0), ]
#- PBMC
identical(colnames(LiveCells_PBMC_UMI), rownames(LiveCells_PBMC_UMI_summary)) # TRUE
LiveCells_PBMC_UMI_filtered <- LiveCells_PBMC_UMI[, which(LiveCells_PBMC_UMI_summary$NUM_GENES >= 400 & LiveCells_PBMC_UMI_summary$NUM_TRANSCRIPTS >= 750)] 
LiveCells_PBMC_UMI_filtered <- LiveCells_PBMC_UMI_filtered[which(apply(LiveCells_PBMC_UMI_filtered, 1, sum) != 0), ] 
#- Merging
LiveCells_UMI <- rbind.fill(as.data.frame(t(LiveCells_LN_UMI_filtered)), as.data.frame(t(LiveCells_PBMC_UMI_filtered)))
rownames(LiveCells_UMI) <- c(paste(colnames(LiveCells_LN_UMI_filtered), 'LN', sep = '_'), paste(colnames(LiveCells_PBMC_UMI_filtered), 'PBMC', sep = '_'))
dim(LiveCells_UMI) 
length(unique(c(rownames(LiveCells_LN_UMI_filtered), rownames(LiveCells_PBMC_UMI_filtered)))) # 19,183 genes
LiveCells_UMI[is.na(LiveCells_UMI)] <- 0 # NAs - replace NA by 0
LiveCells_UMI <- as.data.frame(t(LiveCells_UMI))
```

## GC content and TFH sorted cells

```{r}
ensembl = useMart("ensembl",dataset="hsapiens_gene_ensembl")
filters = listFilters(ensembl)
filters[1:5,]
```

```{r}
out <- getBM(attributes = c('external_gene_name', 'percentage_gene_gc_content'),
      filters = 'external_gene_name', 
      values = rownames(TFH_UMI), 
      mart = ensembl)
```

```{r}
gene_ids <- out$external_gene_name
temp_idx2 <- match(gene_ids, rownames(LiveCells_UMI))
temp_idx <- temp_idx2[!is.na(temp_idx2)]
tmp <- TFH_UMI[temp_idx,]
dropouts <- apply(tmp, 1, function(x) return(length(which(x == 0))/length(x)))
```

```{r}
plot(dropouts,  out$percentage_gene_gc_content[which(!is.na(temp_idx2))], xlab = "Dropout rate",
     ylab = "Percentage GC content")
```
```{r}
gc <-  out$percentage_gene_gc_content[which(!is.na(temp_idx2))]
idx1 <- which(dropouts > 0.98)
idx2 <- which(dropouts < .98 & dropouts > 0.95)
idx3 <- which(dropouts < 0.95 & dropouts > 0.90)
idx4 <- which(dropouts > 0.80 & dropouts < 0.90)
idx5 <- which(dropouts < 0.80 )
```

```{r}
gc2 <- c(gc[idx1], gc[idx2], gc[idx3], gc[idx4], gc[idx5])
Condition <- c(rep('>0.98', length(idx1)), 
               rep('<0.98 | > 0.95', length(idx2)),
               rep('<0.95 | > 0.90', length(idx3)),
               rep('<0.90 | > 0.80', length(idx4)),
               rep('<0.80', length(idx5)))
data_ggplot <- data.frame(Condition = Condition, GC = gc2)

plot1 <- ggplot(data = data_ggplot) + geom_jitter(aes(x = Condition, y = GC, color = Condition)) + geom_violin(aes(x = Condition, y = GC, color = Condition)) + labs(x = '', y = 'GC content') + guides(color = 'none')
plot1
```

## GC content and Live cells

```{r}
out <- getBM(attributes = c('external_gene_name', 'percentage_gene_gc_content'),
      filters = 'external_gene_name', 
      values = rownames(LiveCells_UMI), 
      mart = ensembl)
```

```{r}
gene_ids <- out$external_gene_name
temp_idx2 <- match(gene_ids, rownames(LiveCells_UMI))
temp_idx <- temp_idx2[!is.na(temp_idx2)]
tmp <- TFH_UMI[temp_idx,]
dropouts <- apply(tmp, 1, function(x) return(length(which(x == 0))/length(x)))
```

```{r}
plot(dropouts,  out$percentage_gene_gc_content[which(!is.na(temp_idx2))], xlab = "Dropout rate", ylab = "Percentage GC content")
```

```{r}
gc <-  out$percentage_gene_gc_content[which(!is.na(temp_idx2))]
idx1 <- which(dropouts > 0.98)
idx2 <- which(dropouts < .98 & dropouts > 0.95)
idx3 <- which(dropouts < 0.95 & dropouts > 0.90)
idx4 <- which(dropouts > 0.80 & dropouts < 0.90)
idx5 <- which(dropouts < 0.80 )
```

```{r}
gc2 <- c(gc[idx1], gc[idx2], gc[idx3], gc[idx4], gc[idx5])
Condition <- c(rep('>0.98', length(idx1)), 
               rep('<0.98 | > 0.95', length(idx2)),
               rep('<0.95 | > 0.90', length(idx3)),
               rep('<0.90 | > 0.80', length(idx4)),
               rep('<0.80', length(idx5)))
data_ggplot <- data.frame(Condition = Condition, GC = gc2)

plot1 <- ggplot(data = data_ggplot) + geom_jitter(aes(x = Condition, y = GC, color = Condition)) + geom_violin(aes(x = Condition, y = GC, color = Condition)) + labs(x = '', y = 'GC content') + guides(color = 'none')
plot1
```

