---
title: "singlecell-clustering"
output:
  html_document:
    toc: false
---



We check for the UMI patterns and dropout rates compared to the characteristics of the genes 
like gene length, GC content etc.

## Exploration of scRNA-seq data - UMI, dropouts etc.

[UMI study for TFH146](umi_study_TFH.html)

[Dropout rates versus position of the genes along chromosome and pathway groups](dropout_study.html)

[Dropout rates versus GC content of the genes](dropout_gc_study.html)

We next perform an analysis of the dispersion of genes against the average expression using Seurat and do a pathway based enrichment analysis of the genes that seem to be outliers in this plot.

[Dispersion versus Average expression analysis](cvplot_analysis.html)

## CountClust analysis of the joint LN and PBMC cells for Live cells and TFH sorted cells

We apply CountClust Grade of Membership (GoM) model for clustering the single cell data from lymph 
node (LN) and peripheral blood mononuclear cells (PBMC) of three HIV- and one HIV+ patients.

An example of countclust fit on single cell data from one patient (TFH146) on just the Live cells.

[Example application of CountClust](ex_countclust.html)

[CountClust on TFH146 scRNA-seq data](test_script.html)

[CountClust on TFH041 scRNA-seq data](tfh041Explore.html)

[CountClust on TFH095 scRNA-seq data](tfh095Explore.html)

[CountClust on MP046 scRNA-seq data](mp046Explore.html)


## Separate CountClust analysis of the LN and PBMC cells 

Separate Grade of Membership model fit and subsequent gene annotations and gene set enrichment
analysis of the clusters, for the different patients

First, we perform this analysis for TFH146

[CountClust application separately on LN and PBMC Live cells for TFH146](separate_countclust_LN_PBMC_Live_TFH146.html)

[CountClust application separately on LN and PBMC TFH cells for TFH146](separate_countclust_LN_PBMC_TFH_TFH146.html)

Next we perform the analysis for TFH041

[CountClust application separately on LN and PBMC Live cells for TFH041](separate_countclust_LN_PBMC_Live_TFH041.html)

[CountClust application separately on LN and PBMC TFH cells for TFH041](separate_countclust_LN_PBMC_TFH_TFH041.html)

Finally the analysis on the Live cells for the TFH095 and MP046

[CountClust application separately on LN and PBMC Live cells for TFH095](separate_countclust_LN_PBMC_Live_TFH095.html)

[CountClust application separately on LN and PBMC Live cells for MP046](separate_countclust_LN_PBMC_Live_MP046.html)


## Comparison of CountClust results with t-SNE

We projected the expression of top driving genes from fitting the CountClust model on the t-SNE space obtained using the Seurat package by Satijalab.

The projections for TFH146, TFH041 and TFH095  are 

[t-SNE projection of CountClust top cluster driving genes - TFH 146](tsne_countclust_TFH146.html)

[t-SNE projection of CountClust top cluster driving genes - TFH 095](tsne_countclust_TFH095.html)

[t-SNE projection of CountClust top cluster driving genes - TFH 041](tsne_countclust_TFH041.html)




