---
NEtitle: "t-SNE vs CountClust comparisons - TFH041 Live cells"
author: "Kushal K Dey"
date: "7/28/2017"
output: html_document
---

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE}
library(Seurat)
library(plyr)
library(dplyr)
library(Matrix)
library(cowplot)
library(stringr)
library(MAST)
library(doMC)
library(pheatmap)
library(grid)
library(stringr)
```


## Processing the Data

```{r echo=TRUE, eval=TRUE, message=FALSE, warning=FALSE}
###--- Live Cells
##- LN - no reads information
LiveCells_LN_UMI <- read.table('../data/TFH041/Live Cells/LN/reads.TFH041_Live_LN_S1_R1_001.fastq_bq10_star_corrected.umi.dge.txt', header = TRUE, row.names = 1)
LiveCells_LN_UMI_summary <- read.table('../data/TFH041/Live Cells/LN/reads.TFH041_Live_LN_S1_R1_001.fastq_bq10_star_corrected.umi.dge.summary.txt', header = TRUE, row.names = 1)
#- First filtering - remove cells with less than 10 genes
LiveCells_LN_UMI <- LiveCells_LN_UMI[, rownames(LiveCells_LN_UMI_summary)[which(LiveCells_LN_UMI_summary$NUM_GENES > 10)]]
LiveCells_LN_UMI_summary <- LiveCells_LN_UMI_summary[colnames(LiveCells_LN_UMI), ]

##- PBMC - no reads information
LiveCells_PBMC_UMI <- read.table('../data/TFH041/Live Cells/PBMC/reads.TFH041_Live_PBMC_S2_R1_001.fastq_bq10_star_corrected.umi.dge.txt', header = TRUE, row.names = 1)
LiveCells_PBMC_UMI_summary <- read.table('../data/TFH041/Live Cells/PBMC/reads.TFH041_Live_PBMC_S2_R1_001.fastq_bq10_star_corrected.umi.dge.summary.txt', header = TRUE, row.names = 1)
#- First filtering - remove cells with less than 10 genes
LiveCells_PBMC_UMI <- LiveCells_PBMC_UMI[, rownames(LiveCells_PBMC_UMI_summary)[which(LiveCells_PBMC_UMI_summary$NUM_GENES > 10)]]
LiveCells_PBMC_UMI_summary <- LiveCells_PBMC_UMI_summary[colnames(LiveCells_PBMC_UMI), ]



###--- TFH Cells
##- LN
TFH_LN_UMI <- read.table('../data/TFH041/TFH Sorted/LN/reads.TFH041_TFH_LN_S1_R1_001.fastq_bq10_star_corrected.umi.dge.txt', header = TRUE, row.names = 1)
TFH_LN_reads <- read.table('../data/TFH041/TFH Sorted/LN/reads.TFH041_TFH_LN_S1_R1_001.fastq_bq10_star_corrected.reads.dge.txt', header = TRUE, row.names = 1)
TFH_LN_UMI_summary <- read.table('../data/TFH041/TFH Sorted/LN/reads.TFH041_TFH_LN_S1_R1_001.fastq_bq10_star_corrected.umi.dge.summary.txt', header = TRUE, row.names = 1)
TFH_LN_reads_summary <- read.table('../data/TFH041/TFH Sorted/LN/reads.TFH041_TFH_LN_S1_R1_001.fastq_bq10_star_corrected.reads.dge.summary.txt', header = TRUE, row.names = 1)
#- First filtering - remove cells with less than 10 genes
TFH_LN_UMI <- TFH_LN_UMI[, rownames(TFH_LN_reads_summary)[which(TFH_LN_reads_summary$NUM_GENES > 10)]]
TFH_LN_reads <- TFH_LN_reads[, rownames(TFH_LN_reads_summary)[which(TFH_LN_reads_summary$NUM_GENES > 10)]]
TFH_LN_UMI_summary <- TFH_LN_UMI_summary[colnames(TFH_LN_UMI), ]
TFH_LN_reads_summary <- TFH_LN_reads_summary[colnames(TFH_LN_reads), ]

##- PBMC
TFH_PBMC_UMI <- read.table('../data/TFH041/TFH Sorted/PBMC/reads.TFH041_TFH_PBMC_S2_R1_001.fastq_bq10_star_corrected.umi.dge.txt', header = TRUE, row.names = 1)
TFH_PBMC_reads <- read.table('../data/TFH041/TFH Sorted/PBMC/reads.TFH041_TFH_PBMC_S2_R1_001.fastq_bq10_star_corrected.reads.dge.txt', header = TRUE, row.names = 1)
TFH_PBMC_UMI_summary <- read.table('../data/TFH041/TFH Sorted/PBMC/reads.TFH041_TFH_PBMC_S2_R1_001.fastq_bq10_star_corrected.umi.dge.summary.txt', header = TRUE, row.names = 1)
TFH_PBMC_reads_summary <- read.table('../data/TFH041/TFH Sorted/PBMC/reads.TFH041_TFH_PBMC_S2_R1_001.fastq_bq10_star_corrected.reads.dge.summary.txt', header = TRUE, row.names = 1)
#- First filtering - remove cells with less than 10 genes
TFH_PBMC_UMI <- TFH_PBMC_UMI[, rownames(TFH_PBMC_reads_summary)[which(TFH_PBMC_reads_summary$NUM_GENES > 10)]]
TFH_PBMC_reads <- TFH_PBMC_reads[, rownames(TFH_PBMC_reads_summary)[which(TFH_PBMC_reads_summary$NUM_GENES > 10)]]
TFH_PBMC_UMI_summary <- TFH_PBMC_UMI_summary[colnames(TFH_PBMC_UMI), ]
TFH_PBMC_reads_summary <- TFH_PBMC_reads_summary[colnames(TFH_PBMC_reads), ]
```

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE}
###--- Filtering datasets and merging
#- LN
identical(colnames(TFH_LN_UMI), rownames(TFH_LN_UMI_summary)) # TRUE
TFH_LN_UMI_filtered <- TFH_LN_UMI[, which(TFH_LN_UMI_summary$NUM_GENES >= 400 & TFH_LN_UMI_summary$NUM_TRANSCRIPTS >= 750)] # 344 cells
TFH_LN_UMI_filtered <- TFH_LN_UMI_filtered[which(apply(TFH_LN_UMI_filtered, 1, sum) != 0), ] # 14,856 genes and 344 cells
#- PBMC
identical(colnames(TFH_PBMC_UMI), rownames(TFH_PBMC_UMI_summary)) # TRUE
TFH_PBMC_UMI_filtered <- TFH_PBMC_UMI[, which(TFH_PBMC_UMI_summary$NUM_GENES >= 400 & TFH_PBMC_UMI_summary$NUM_TRANSCRIPTS >= 750)] # 361 cells
TFH_PBMC_UMI_filtered <- TFH_PBMC_UMI_filtered[which(apply(TFH_PBMC_UMI_filtered, 1, sum) != 0), ] # 14,657 genes and 361 cells
#- Merging
TFH_UMI <- rbind.fill(as.data.frame(t(TFH_LN_UMI_filtered)), as.data.frame(t(TFH_PBMC_UMI_filtered)))
rownames(TFH_UMI) <- c(paste(colnames(TFH_LN_UMI_filtered), 'LN', sep = '_'), paste(colnames(TFH_PBMC_UMI_filtered), 'PBMC', sep = '_'))
dim(TFH_UMI) # 705 cells and 17,772 genes
length(unique(c(rownames(TFH_LN_UMI_filtered), rownames(TFH_PBMC_UMI_filtered)))) # 19,384 genes
TFH_UMI[is.na(TFH_UMI)] <- 0 # NAs - replace NA by 0
TFH_UMI <- as.data.frame(t(TFH_UMI))
```


## TFH - (tSNE + CountClust genes)

```{r echo=FALSE, eval=FALSE}
seuratObj <- new('seurat', raw.data = TFH_UMI)
seuratObj <- Setup(seuratObj, min.cells = 0, min.genes = 0, project = 'Pantaleo', total.expr = 10000) # no need to filter - already done
#- Output
seuratObj_TFH_global <- seuratObj
```


```{r echo=TRUE, eval=FALSE, message=FALSE, warning=FALSE}
###--- Regress out unwanted sources of variation
mito.genes <- grep("^MT-", rownames(seuratObj_TFH_global@data), value = TRUE)
percent.mito <- colSums(as.matrix(expm1(seuratObj_TFH_global@data[mito.genes, ]))) / colSums(as.matrix(expm1(seuratObj_TFH_global@data)))
seuratObj_TFH_global <- AddMetaData(seuratObj_TFH_global, percent.mito, 'percent.mito')
GenePlot(seuratObj_TFH_global, 'nUMI', 'percent.mito')
GenePlot(seuratObj_TFH_global, 'nUMI', 'nGene')
#- remove unwanted sources of variation
seuratObj_TFH_global <- RegressOut(seuratObj_TFH_global, latent.vars = c('nUMI', 'percent.mito')) # remove effect of CDR (high correlation between nGene and nUMI - 0.97) and mitochondrial
```


```{r echo=TRUE, eval=FALSE}
seuratObj_TFH_global <- MeanVarPlot(seuratObj_TFH_global, fxn.x = expMean, fxn.y = logVarDivMean, x.low.cutoff = 0.25, x.high.cutoff = 3, y.cutoff = 0.5, do.contour = FALSE)
length(seuratObj_TFH_global@var.genes)
seuratObj_TFH_global <- PCA(seuratObj_TFH_global, pc.genes = seuratObj_TFH_global@var.genes)
seuratObj_TFH_global <- ProjectPCA(seuratObj_TFH_global)
VizPCA(seuratObj_TFH_global, 1:2)
```

```{r echo=TRUE, eval=FALSE}
seuratObj_TFH_global <- RunTSNE(seuratObj_TFH_global, dims.use = 1:10, perplexity = 50)
```

```{r echo=FALSE, eval=FALSE}
save(seuratObj_TFH_global, file = "../output/TFH041/seuratTFH.rda")
```

```{r}
seuratObj_TFH_global <- get(load("../output/TFH041/seuratTFH.rda"))
```

```{r echo=TRUE, eval=TRUE}
seuratObj_TFH_global <- MeanVarPlot(seuratObj_TFH_global, fxn.x = expMean, fxn.y = logVarDivMean, x.low.cutoff = 0.25, x.high.cutoff = 3, y.cutoff = 0.5, do.contour = FALSE)
length(seuratObj_TFH_global@var.genes)
seuratObj_TFH_global <- PCA(seuratObj_TFH_global, pc.genes = seuratObj_TFH_global@var.genes)
seuratObj_TFH_global <- ProjectPCA(seuratObj_TFH_global)
VizPCA(seuratObj_TFH_global, 1:2)
```

```{r}
###--- tSNE
labels <- unlist(str_split(rownames(seuratObj_TFH_global@tsne.rot), '_'))[seq(2, length(unlist(str_split(rownames(seuratObj_TFH_global@tsne.rot), '_'))), 2)]
nUMI <- seuratObj_TFH_global@data.info$nUMI
nGene <- seuratObj_TFH_global@data.info$nGene

data_ggplot <- data.frame(labels = labels,
                          nUMI = nUMI,
                          nGene = nGene,
                          tSNE_1 = seuratObj_TFH_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_TFH_global@tsne.rot[, 2])

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = labels), size = 0.5) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_color_discrete('')
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = nUMI), size = 0.5) + labs(x = 'tSNE 1', y = 'tSNE 2') + scale_color_gradient2('# mapped UMI', low = 'royalblue', high = 'red', mid = 'grey80', midpoint = 2000)
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = nGene), size = 0.5) + labs(x = 'tSNE 1', y = 'tSNE 2') + scale_color_gradient2('# detected genes', low = 'royalblue', high = 'red', mid = 'grey80', midpoint = 1000)

plot_grid(plot1, plot2, plot3, ncol = 2, align = 'v')
```










We now use the top genes from cluster 1 and see where their expression values lie in the t-SNE plot.

### Cluster 1

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, fig.width=15, fig.height=16}
###--- tSNE - CountClust - cluster 1
cex.size = 2
data_ggplot <- data.frame(tSNE_1 = seuratObj_TFH_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_TFH_global@tsne.rot[, 2],
                          TPT1 = seuratObj_TFH_global@data['TPT1', ], 
                          BTF3 = seuratObj_TFH_global@data['BTF3', ],
                          SNHG5 = seuratObj_TFH_global@data['SNHG5', ], # TFH marker
                          EEF1A1 = seuratObj_TFH_global@data['EEF1A1', ] # TFH marker
                          ) # effector t cell

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = TPT1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['TPT1', ], 0.2),  name = "TPT1")
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = BTF3), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['BTF3', ], 0.2),  name = "BTF3")
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = SNHG5), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['SNHG5', ], 0.2),  name = "SNHG5")
plot4 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = EEF1A1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['EEF1A1', ], 0.2),  name = "EEF1A1")

plot_grid(plot1, plot2, plot3, plot4, ncol = 2, align = 'v')
```





### Cluster 2

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, fig.width=15, fig.height=16}
###--- tSNE - CountClust - cluster 1
cex.size = 2
data_ggplot <- data.frame(tSNE_1 = seuratObj_TFH_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_TFH_global@tsne.rot[, 2],
                          MS4A1 = seuratObj_TFH_global@data['MS4A1', ], 
                          HLADRA = seuratObj_TFH_global@data['HLA-DRA', ],
                          MALAT1 = seuratObj_TFH_global@data['MALAT1', ], # TFH marker
                          IGHM = seuratObj_TFH_global@data['IGHM', ] # TFH marker
                          ) # effector t cell

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = MS4A1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['MS4A1', ], 0.2),  name = "MS4A1")
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = HLADRA), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['HLA-DRA', ], 0.2),  name = "HLA-DRA")
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = MALAT1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['MALAT1', ], 0.2),  name = "MALAT1")
plot4 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = IGHM), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['IGHM', ], 0.2),  name = "IGHM")

plot_grid(plot1, plot2, plot3, plot4, ncol = 2, align = 'v')
```



### Cluster 3

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, fig.width=15, fig.height=16}
###--- tSNE - CountClust - cluster 3
cex.size = 2
data_ggplot <- data.frame(tSNE_1 = seuratObj_TFH_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_TFH_global@tsne.rot[, 2],
                          IL7R = seuratObj_TFH_global@data['IL7R', ], 
                          CD3D = seuratObj_TFH_global@data['CD3D', ],
                          TRBC2 = seuratObj_TFH_global@data['TRBC2', ], # TFH marker
                          SRGN = seuratObj_TFH_global@data['SRGN', ] # TFH marker
                          ) # effector t cell

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = IL7R), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['IL7R', ], 0.2),  name = "IL7R")
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = CD3D), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['CD3D', ], 0.2),  name = "CD3D")
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = TRBC2), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['TRBC2', ], 0.2),  name = "TRBC2")
plot4 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = SRGN), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['SRGN', ], 0.2),  name = "SRGN")

plot_grid(plot1, plot2, plot3, plot4, ncol = 2, align = 'v')
```


### Cluster 4

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, fig.width=15, fig.height=16}
###--- tSNE - CountClust - cluster 4
cex.size = 2
data_ggplot <- data.frame(tSNE_1 = seuratObj_TFH_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_TFH_global@tsne.rot[, 2],
                          IGKC = seuratObj_TFH_global@data['IGKC', ], 
                          IGHG2 = seuratObj_TFH_global@data['IGHG2', ],
                          IGKV41 = seuratObj_TFH_global@data['IGKV4-1', ], # TFH marker
                          IGHG1 = seuratObj_TFH_global@data['IGHG1', ] # TFH marker
                          ) # effector t cell

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = IGKC), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['IGKC', ], 0.2),  name = "IGKC")
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = IGHG2), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['IGHG2', ], 0.2),  name = "IGHG2")
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = IGKV41), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['IGKV4-1', ], 0.2),  name = "IGKV4-1")
plot4 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = IGHG1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_TFH_global@data['IGHG1', ], 0.2),  name = "IGHG1")

plot_grid(plot1, plot2, plot3, plot4, ncol = 2, align = 'v')
```


## Live cells clustering 

```{r echo=TRUE, eval=TRUE, message=FALSE, warning=FALSE}
###--- Filtering datasets and merging
#- LN
identical(colnames(LiveCells_LN_UMI), rownames(LiveCells_LN_UMI_summary)) # TRUE
LiveCells_LN_UMI_filtered <- LiveCells_LN_UMI[, which(LiveCells_LN_UMI_summary$NUM_GENES >= 400 & LiveCells_LN_UMI_summary$NUM_TRANSCRIPTS >= 750)] # 417 cells
LiveCells_LN_UMI_filtered <- LiveCells_LN_UMI_filtered[which(apply(LiveCells_LN_UMI_filtered, 1, sum) != 0), ] # 14,946 genes and 417 cells
#- PBMC
identical(colnames(LiveCells_PBMC_UMI), rownames(LiveCells_PBMC_UMI_summary)) # TRUE
LiveCells_PBMC_UMI_filtered <- LiveCells_PBMC_UMI[, which(LiveCells_PBMC_UMI_summary$NUM_GENES >= 400 & LiveCells_PBMC_UMI_summary$NUM_TRANSCRIPTS >= 750)] # 409 cells
LiveCells_PBMC_UMI_filtered <- LiveCells_PBMC_UMI_filtered[which(apply(LiveCells_PBMC_UMI_filtered, 1, sum) != 0), ] # 12,391 genes and 409 cells
#- Merging
LiveCells_UMI <- rbind.fill(as.data.frame(t(LiveCells_LN_UMI_filtered)), as.data.frame(t(LiveCells_PBMC_UMI_filtered)))
rownames(LiveCells_UMI) <- c(paste(colnames(LiveCells_LN_UMI_filtered), 'LN', sep = '_'), paste(colnames(LiveCells_PBMC_UMI_filtered), 'PBMC', sep = '_'))
dim(LiveCells_UMI) # 826 cells and 16,230 genes
length(unique(c(rownames(LiveCells_LN_UMI_filtered), rownames(LiveCells_PBMC_UMI_filtered)))) # 16,230 genes
LiveCells_UMI[is.na(LiveCells_UMI)] <- 0 # NAs - replace NA by 0
LiveCells_UMI <- as.data.frame(t(LiveCells_UMI))
```


## Live Cells - (tSNE + CountClust genes)


```{r echo=FALSE, eval=FALSE}
seuratObj <- new('seurat', raw.data = LiveCells_UMI)
seuratObj <- Setup(seuratObj, min.cells = 0, min.genes = 0, project = 'Pantaleo', total.expr = 10000) # no need to filter - already done
#- Output
seuratObj_LiveCells_global <- seuratObj
```

```{r echo=FALSE, eval=FALSE, message=FALSE, warning=FALSE}
###--- Regress out unwanted sources of variation
mito.genes <- grep("^MT-", rownames(seuratObj_LiveCells_global@data), value = TRUE)
percent.mito <- colSums(as.matrix(expm1(seuratObj_LiveCells_global@data[mito.genes, ]))) / colSums(as.matrix(expm1(seuratObj_LiveCells_global@data)))
seuratObj_LiveCells_global <- AddMetaData(seuratObj_LiveCells_global, percent.mito, 'percent.mito')
GenePlot(seuratObj_LiveCells_global, 'nUMI', 'percent.mito')
GenePlot(seuratObj_LiveCells_global, 'nUMI', 'nGene')
#- remove unwanted sources of variation
seuratObj_LiveCells_global <- RegressOut(seuratObj_LiveCells_global, latent.vars = c('nUMI', 'percent.mito')) # remove effect of CDR (high correlation between nGene and nUMI - 0.97) and mitochondrial
```

```{r echo=FALSE, eval=FALSE}
seuratObj_LiveCells_global <- MeanVarPlot(seuratObj_LiveCells_global, fxn.x = expMean, fxn.y = logVarDivMean, x.low.cutoff = 0.25, x.high.cutoff = 3, y.cutoff = 0.5, do.contour = FALSE)
length(seuratObj_LiveCells_global@var.genes)
seuratObj_LiveCells_global <- PCA(seuratObj_LiveCells_global, pc.genes = seuratObj_LiveCells_global@var.genes)
seuratObj_LiveCells_global <- ProjectPCA(seuratObj_LiveCells_global)
VizPCA(seuratObj_LiveCells_global, 1:2)
```

```{r echo=FALSE, eval=FALSE}
seuratObj_LiveCells_global <- RunTSNE(seuratObj_LiveCells_global, dims.use = 1:10, perplexity = 50)
```

```{r echo=FALSE, eval=FALSE}
save(seuratObj_LiveCells_global, file = "../output/TFH041/seuratLiveCells.rda")
```

```{r}
seuratObj_LiveCells_global <- get(load("../output/TFH041/seuratLiveCells.rda"))
```


```{r echo=TRUE, eval=TRUE}
seuratObj_LiveCells_global <- MeanVarPlot(seuratObj_LiveCells_global, fxn.x = expMean, fxn.y = logVarDivMean, x.low.cutoff = 0.25, x.high.cutoff = 3, y.cutoff = 0.5, do.contour = FALSE)
length(seuratObj_LiveCells_global@var.genes)
seuratObj_LiveCells_global <- PCA(seuratObj_LiveCells_global, pc.genes = seuratObj_LiveCells_global@var.genes)
seuratObj_LiveCells_global <- ProjectPCA(seuratObj_LiveCells_global)
VizPCA(seuratObj_LiveCells_global, 1:2)
```

```{r}
###--- tSNE
labels <- unlist(str_split(rownames(seuratObj_LiveCells_global@tsne.rot), '_'))[seq(2, length(unlist(str_split(rownames(seuratObj_LiveCells_global@tsne.rot), '_'))), 2)]
nUMI <- seuratObj_LiveCells_global@data.info$nUMI
nGene <- seuratObj_LiveCells_global@data.info$nGene

data_ggplot <- data.frame(labels = labels,
                          nUMI = nUMI,
                          nGene = nGene,
                          tSNE_1 = seuratObj_LiveCells_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_LiveCells_global@tsne.rot[, 2])

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = labels), size = 0.5) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_color_discrete('')
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = nUMI), size = 0.5) + labs(x = 'tSNE 1', y = 'tSNE 2') + scale_color_gradient2('# mapped UMI', low = 'royalblue', high = 'red', mid = 'grey80', midpoint = 2000)
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = nGene), size = 0.5) + labs(x = 'tSNE 1', y = 'tSNE 2') + scale_color_gradient2('# detected genes', low = 'royalblue', high = 'red', mid = 'grey80', midpoint = 1000)

plot_grid(plot1, plot2, plot3, ncol = 2, align = 'v')
```

### Cluster 1

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, fig.width=15, fig.height=16}
###--- tSNE - CountClust - cluster 1
cex.size = 2
data_ggplot <- data.frame(tSNE_1 = seuratObj_LiveCells_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_LiveCells_global@tsne.rot[, 2],
                          TMSB4X = seuratObj_LiveCells_global@data['TMSB4X', ], 
                          TMSB10 = seuratObj_LiveCells_global@data['TMSB10', ],
                          GNB2L1 = seuratObj_LiveCells_global@data['GNB2L1', ], # LiveCells marker
                          EEF1A1 = seuratObj_LiveCells_global@data['EEF1A1', ] # LiveCells marker
                          ) # effector t cell

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = TMSB4X), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['TMSB4X', ], 0.2),  name = "TMSB4X")
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = TMSB10), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['TMSB10', ], 0.2),  name = "TMSB10")
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = GNB2L1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['GNB2L1', ], 0.2),  name = "GNB2L1")
plot4 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = EEF1A1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['EEF1A1', ], 0.2),  name = "EEF1A1")

plot_grid(plot1, plot2, plot3, plot4, ncol = 2, align = 'v')
```


### Cluster 2

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, fig.width=15, fig.height=16}
###--- tSNE - CountClust - cluster 1
cex.size = 2
data_ggplot <- data.frame(tSNE_1 = seuratObj_LiveCells_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_LiveCells_global@tsne.rot[, 2],
                          TPT1 = seuratObj_LiveCells_global@data['TPT1', ], 
                          TXNIP = seuratObj_LiveCells_global@data['TXNIP', ],
                          IL7R = seuratObj_LiveCells_global@data['IL7R', ], # LiveCells marker
                          CALM1 = seuratObj_LiveCells_global@data['CALM1', ] # LiveCells marker
                          ) # effector t cell

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = TPT1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['TPT1', ], 0.2),  name = "TPT1")
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = TXNIP), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['TXNIP', ], 0.2),  name = "TXNIP")
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = IL7R), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['IL7R', ], 0.2),  name = "IL7R")
plot4 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = CALM1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['CALM1', ], 0.2),  name = "CALM1")

plot_grid(plot1, plot2, plot3, plot4, ncol = 2, align = 'v')
```

### Cluster 3

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, fig.width=15, fig.height=16}
###--- tSNE - CountClust - cluster 1
cex.size = 2
data_ggplot <- data.frame(tSNE_1 = seuratObj_LiveCells_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_LiveCells_global@tsne.rot[, 2],
                          MALAT1 = seuratObj_LiveCells_global@data['MALAT1', ], 
                          SRGN = seuratObj_LiveCells_global@data['SRGN', ],
                          CD3D = seuratObj_LiveCells_global@data['CD3D', ], # LiveCells marker
                          ITM2A = seuratObj_LiveCells_global@data['ITM2A', ] # LiveCells marker
                          ) # effector t cell

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = MALAT1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['MALAT1', ], 0.2),  name = "MALAT1")
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = SRGN), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['SRGN', ], 0.2),  name = "SRGN")
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = CD3D), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['CD3D', ], 0.2),  name = "CD3D")
plot4 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = ITM2A), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['ITM2A', ], 0.2),  name = "ITM2A")

plot_grid(plot1, plot2, plot3, plot4, ncol = 2, align = 'v')
```


### Cluster 4

```{r echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE, fig.width=15, fig.height=16}
###--- tSNE - CountClust - cluster 1
cex.size = 2
data_ggplot <- data.frame(tSNE_1 = seuratObj_LiveCells_global@tsne.rot[, 1],
                          tSNE_2 = seuratObj_LiveCells_global@tsne.rot[, 2],
                          ACTB = seuratObj_LiveCells_global@data['ACTB', ], 
                          MYL6 = seuratObj_LiveCells_global@data['MYL6', ],
                          PFN1 = seuratObj_LiveCells_global@data['PFN1', ], # LiveCells marker
                          GAPDH = seuratObj_LiveCells_global@data['GAPDH', ] # LiveCells marker
                          ) # effector t cell

plot1 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = ACTB), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['ACTB', ], 0.2),  name = "ACTB")
plot2 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = MYL6), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['MYL6', ], 0.2),  name = "MYL6")
plot3 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = PFN1), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['PFN1', ], 0.2),  name = "PFN1")
plot4 <- ggplot(data_ggplot) + theme(panel.background = element_rect(colour = "black", size = 1), 
                                     panel.grid.major = element_blank(), 
                                     panel.grid.minor = element_blank(), 
                                     axis.text.y = element_blank(),
                                     axis.text.x = element_blank(),
                                     axis.ticks = element_blank(),
                                     legend.text = element_text(size = 15)) + geom_point(aes(x = tSNE_1, y = tSNE_2, color = GAPDH), size = cex.size) + labs(x = 'tSNE 1', y = 'tSNE 2', title = '') + scale_colour_gradient2(low = 'royalblue', high = 'red', mid = 'grey80', midpoint = quantile(seuratObj_LiveCells_global@data['GAPDH', ], 0.2),  name = "GAPDH")

plot_grid(plot1, plot2, plot3, plot4, ncol = 2, align = 'v')
```
